pkg/cloud/aws/services/ec2/securitygroups.go

```diff
@@ -334,34 +334,63 @@ func (s *Service) getSecurityGroupTagParams(name string, role v1alpha1.SecurityG
 }
 
 func ingressRuleToSDKType(i *v1alpha1.IngressRule) *ec2.IpPermission {
-	res := &ec2.IpPermission{
-		IpProtocol: aws.String(string(i.Protocol)),
-		FromPort:   aws.Int64(i.FromPort),
-		ToPort:     aws.Int64(i.ToPort),
+	var res *ec2.IpPermission
+	switch i.Protocol {
+	case v1alpha1.SecurityGroupProtocolTCP,
+		v1alpha1.SecurityGroupProtocolUDP,
+		v1alpha1.SecurityGroupProtocolICMP,
+		v1alpha1.SecurityGroupProtocolICMPv6:
+		res = &ec2.IpPermission{
+			IpProtocol: aws.String(string(i.Protocol)),
+			FromPort:   aws.Int64(i.FromPort),
+			ToPort:     aws.Int64(i.ToPort),
+		}
+	default:
+		res = &ec2.IpPermission{
+			IpProtocol: aws.String(string(i.Protocol)),
+		}
 	}
 
 	for _, cidr := range i.CidrBlocks {
-		res.IpRanges = append(res.IpRanges, &ec2.IpRange{
-			Description: aws.String(i.Description),
-			CidrIp:      aws.String(cidr),
-		})
+		ipRange := &ec2.IpRange{
+			CidrIp: aws.String(cidr),
+		}
+
+		if i.Description != "" {
+			ipRange.Description = aws.String(i.Description)
+		}
+
+		res.IpRanges = append(res.IpRanges, ipRange)
 	}
 
 	for _, groupID := range i.SourceSecurityGroupIDs {
-		res.UserIdGroupPairs = append(res.UserIdGroupPairs, &ec2.UserIdGroupPair{
-			Description: aws.String(i.Description),
-			GroupId:     aws.String(groupID),
-		})
+		userIDGroupPair := &ec2.UserIdGroupPair{
+			GroupId: aws.String(groupID),
+		}
+
+		if i.Description != "" {
+			userIDGroupPair.Description = aws.String(i.Description)
+		}
+
+		res.UserIdGroupPairs = append(res.UserIdGroupPairs, userIDGroupPair)
 	}
 
 	return res
 }
 
 func ingressRuleFromSDKType(v *ec2.IpPermission) *v1alpha1.IngressRule {
-	res := &v1alpha1.IngressRule{
-		Protocol: v1alpha1.SecurityGroupProtocol(*v.IpProtocol),
-		FromPort: *v.FromPort,
-		ToPort:   *v.ToPort,
+	var res *v1alpha1.IngressRule
+	switch *v.IpProtocol {
+	case "tcp", "udp", "icmp", "58" /* ICMPv6 */ :
```

Comment:
Should we move all of these to be package constants with descriptions?

Reply:
You don't like the `"58"` hanging out there, apropos of nothing? 😆 

I'll move 'em on in

Reply:
That for sure! 😂 Main reasoning was that these variables are probably going to be reused in other places too so it might be worthwhile to declare them somewhere else.